Python/optimizer_bytecodes.c

```diff
@@ -957,6 +957,16 @@ dummy_func(void) {
         }
     }
 
+    op(_CALL_TUPLE_1, (callable, null, arg -- res)) {
+        if (sym_matches_type(arg, &PyTuple_Type)) {
+            // e.g. tuple((1, 2)) or tuple(foo) where foo is known to be a tuple
+            res = arg;
+        }
```

Comment:
Same optimization as in https://github.com/python/cpython/pull/132849#discussion_r2059012758